src/transfer/stores/headless.ts

```diff
@@ -80,35 +96,117 @@ export default class HeadlessStore implements IHeadlessStore {
 
   @action
   transferGold = async (
+    signer: string,
     recipient: string,
     amount: Decimal,
     memo: string
   ): Promise<TxId> => {
-    this.assertAgentAddress();
+    if (signer.startsWith("0x")) {
+      signer = signer.substr(2);
+    }
+    if (recipient.startsWith("0x")) {
+      recipient = recipient.substr(2);
+    }
 
-    const nextTxNonceData = await this.graphqlSdk.GetNextTxNonce({
-      address: this.agentAddress,
-    });
-    if (!nextTxNonceData.data) {
-      throw new Error("Failed to get next nonce");
+    if (signer === recipient) {
+      const errorMessage = "You can't transfer NCG to yourself.";
+      alert(errorMessage);
+      throw new Error(errorMessage);
     }
-    const txNonce = nextTxNonceData.data.transaction.nextTxNonce as string;
-    const tx = await this.graphqlSdk.Transfer({
+
+    this.assertAgentAddressV2(signer);
+
+    async function makeTx(
+      signer: string,
+      recipient: string,
+      amount: Decimal,
+      memo: string,
+      fileName: string,
+      graphqlSdk: GraphQLSDK
+    ) {
+      // create action.
+      if (
+        !ipcRenderer.sendSync(
+          "transfer-asset",
+          signer,
+          recipient,
+          Number(amount),
+          memo,
+          fileName
+        )
+      ) {
+        return;
+      }
+
+      // get tx nonce.
+      const ended = async (signer: string, graphqlSdk: GraphQLSDK) => {
+        return await graphqlSdk.GetNextTxNonce({
+          address: signer,
+        });
+      };
+
+      let txNonce;
+      try {
+        let res = await ended(signer, graphqlSdk);
+        txNonce = res.data?.transaction.nextTxNonce;
+      } catch (e) {
+        alert(e.message);
+        return;
+      }
+
+      // sign tx.
+      const result = ipcRenderer.sendSync(
+        "sign-tx",
+        txNonce,
+        new Date().toISOString(),
+        fileName
+      );
+
+      if (result.stderr != "") {
+        alert(result.stderr);
+        return;
+      }
+
+      return result.stdout as string;
+    }
+
+    const fileName = await tmpName();
+    const tx = await makeTx(
+      signer,
       recipient,
-      amount: amount.toString(),
+      amount,
       memo,
-      txNonce,
-    });
-    if (!tx.data) {
-      throw new Error(`Failed to create transaction. ${tx.errors}`);
+      fileName,
+      this.graphqlSdk
+    );
+
+    if (tx == undefined) {
+      alert("failed tx creation.");
+      throw new Error("Failed to create transaction.");
     }
 
-    return tx.data.transfer;
+    const transferResult = await this.graphqlSdk.StageTxV2({ encodedTx: tx });
+
+    if (transferResult.data == null) {
```

Comment:
```suggestion
    if (transferResult.data === null) {
```

Reply:
No, this is one of the very few cases where it might be appropriate to use `==`. `a == null` is truthy when a is null or undefined, which is helpful to check two empty values of JS.